x/qgb/keeper/keeper_valset.go

```diff
@@ -95,11 +96,12 @@ func (k Keeper) GetCurrentValset(ctx sdk.Context) (types.Valset, error) {
 
 		evmAddress, exists := k.GetEVMAddress(ctx, val.String())
 		if !exists {
-			// nodes must have an evm address to be held accountable for. Since
-			// the protocol doesn't require that they register it, if they don't
-			// we take some unique string to derive their evm address from, namely
-			// the validators address
-			evmAddress = gethcommon.BytesToAddress(val).String()
+			// This should never happen and indicates a bug in the design of
+			// the system (most likely that a hook wasn't called). A validator
+			// should always have an associated EVM address. Fortunately we can
+			// safely recover from this by deriving the default again.
+			ctx.Logger().Error("validator does not have an evm address set")
+			evmAddress = types.DefaultEVMAddress(val).String()
```

Comment:
If we create a new EVM address here, this is not persisting it in store. Does it make sense to loop for correct default addresses in the hook?
Also, does it make sense to have a safer mechanism to derive the default EVM address? Something like hashing the operator address + something deterministic and specific to the current block?

Reply:
> Also, does it make sense to have a safer mechanism to derive the default EVM address? Something like hashing the operator address + something deterministic and specific to the current block?

I assume the validator address is unique thus it's safe to derive the EVM address from? Is there a reason why you would like to add more variables? 

> If we create a new EVM address here, this is not persisting it in store. Does it make sense to loop for correct default addresses in the hook?

Hmm so in theory this should never happen but perhaps you're right that it's better that if there is a developer error that we still persist the new default EVM address

Reply:
> I assume the validator address is unique thus it's safe to derive the EVM address from? Is there a reason why you would like to add more variables?

the validator address is unique but as was discussed yesterday, some other validator can derive your validator EVM address and use it for themselves. Especially if the validator reuses keys

Reply:
I guess it makes sense to open an issue to keep note that we can have a squatting issue and we need to investigate it when we have time